### PR TITLE
fix: async build plugins

### PR DESCRIPTION
**Pull request by anuraghazra**
DO NOT MERGE, Testing only.

## What type of PR is this?

- [ ] Feature
- [x] Bug Fix
- [ ] Refactor
- [ ] Documentation Update

## Description
related to #36 

## Added unit tests?

- [ ] No, I want help in writing tests.
- [x] No, I want someone else to write tests.
- [ ] No, The addition does not need tests.
- [ ] Yes

### CODE REVIEW

File: src/cli.js [hunk 1]
```diff
@@ -1,19 +1,42 @@
...
+      }
+    }
+
+    /**
+     getBaseProgram again after making sure
+     async operations are done and "contents"  
+     folder have all the data
+    */
+    programInfo = getBaseProgramInfo();
+
```
Comment (anuraghazra) [inline src/cli.js:35]: As you can see i have moved the beforeBuild logic in here. 

on line 15: i get the programInfo that has all the info except the contentDirectories.

Then our plugin runs and it awaits until the async operations are done.

on line 34: i again called getBaseInfoProgram to get the latest info which will include the contentDirectories and after that we will just call the build()

Comment (saurabhdaware) [reply src/cli.js:35]: Yes makes sense, `loadContent` is supposed to work individually though, what issue you faced in it? you have to pass contentPath from abellConfigs
```js
loadContent(programInfo.abellConfigs.contentPath)
```

Comment (anuraghazra) [reply src/cli.js:35]: There was some $path is undefined error.... let me check that again

Comment (anuraghazra) [reply src/cli.js:35]: TypeError: Cannot read property '$path' of undefined
    at generateContentFile (D:\CAM Files\Dreamwaver\Javascript\github\abell\src\utils\build-utils.js:265:31)
    at build (D:\CAM Files\Dreamwaver\Javascript\github\abell\src\commands\build.js:40:9)
    at Command.<anonymous> (D:\CAM Files\Dreamwaver\Javascript\github\abell\src\cli.js:40:5)

Comment (anuraghazra) [reply src/cli.js:35]: And btw its happening only when there is no content & dist folder

Comment (saurabhdaware) [reply src/cli.js:35]: yes there is supposed to be a content folder for it to work,

Comment (saurabhdaware) [reply src/cli.js:35]: loadContent will basically look into all meta.json files inside `content` folder

**Comment by saurabhdaware** [comment]
Hi @anuraghazra, are you working on this? if no then I can take this into separate branch and continue with your code.

**Comment by anuraghazra** [comment]
hi @saurabhdaware, sorry didn't really got the time to look into this further. you can make this into a branch and continute the development if you want :D.

**Comment by saurabhdaware** [comment]
no issues! already a huge help! yes I will continue with your code in separate branch. Thanks!

**Pull request merged**